File: path-sum/path_sum_test.go
package pathSum

import "testing"

func TestHasPathSum(t *testing.T) {
	var testCases = []struct {
		root      *TreeNode
		targetSum int
		want      bool
	}{
		{
			root: &TreeNode{
				Val: 5,
				Left: &TreeNode{
					Val: 4,
					Left: &TreeNode{
						Val:   11,
						Left:  leafNode(7),
						Right: leafNode(2),
					},
					Right: nil,
				},
				Right: &TreeNode{
					Val:  8,
					Left: leafNode(13),
					Right: &TreeNode{
						Val:   4,
						Left:  nil,
						Right: leafNode(1),
					},
				},
			},
			targetSum: 22,
			want:      true,
		},
		{
			root: &TreeNode{
				Val:   1,
				Left:  leafNode(2),
				Right: leafNode(3),
			},
			targetSum: 5,
			want:      false,
		},
		{
			root:      nil,
			targetSum: 0,
			want:      false,
		},
	}

	for _, testCase := range testCases {
		if r := hasPathSum(testCase.root, testCase.targetSum); r != testCase.want {
			t.Errorf("hasPathSum(%v, %d) == %v, expected %v", testCase.root, testCase.targetSum, r, testCase.want)
		}
	}
}

func leafNode(val int) *TreeNode {
	return &TreeNode{
		Val:   val,
		Left:  nil,
		Right: nil,
	}
}

File: tree/path-sum/python/path_sum_test.py
import unittest

from path_sum import TreeNode, Solution

class TestPathSum(unittest.TestCase):
    def setUp(self):
        self.s = Solution()

        self.testCases = [
            {
                'root': TreeNode(
                    5, TreeNode(4,TreeNode(11, TreeNode(7), TreeNode(2)), None), TreeNode(8, TreeNode(13), TreeNode(4, None, TreeNode(1))),
                ),
                'targetSum': 22,
                'want': True,
            },
            {
                'root': TreeNode(
                    1, TreeNode(2), TreeNode(3),
                ),
                'targetSum': 5,
                'want': False,
            },
            {
                'root': TreeNode(
                    1, TreeNode(2), None,
                ),
                'targetSum': 0,
                'want': False,
            },
            {
                'root': TreeNode(
                    1, TreeNode(2), None,
                ),
                'targetSum': 1,
                'want': False,
            },
            {
                'root': None,
                'targetSum': 0,
                'want': False,
            },
        ]


    def test(self):
        for tc in self.testCases:
            with self.subTest(tc=tc):
                self.assertEqual(
                    self.s.hasPathSum(tc['root'], tc['targetSum']),
                    tc['want'],
                )


if __name__ == '__main__':
    unittest.main()
